src/kontrol/prove.py

```diff
@@ -430,6 +440,57 @@ def _method_to_cfg(
     return cfg, new_node_ids, init_node_id, target_node.id
 
 
+def _process_summary(summary: dict) -> dict:
+    accounts: dict[int, dict] = {}
+
+    def _init_account(address: int) -> None:
+        if address not in accounts.keys():
+            accounts[address] = {'balance': 0, 'nonce': 0, 'code': '', 'storage': {}}
+
+    for entry in summary:
+        e = SummaryEntry(entry)
+
+        if e.has_ignored_kind or e.reverted:
+            continue
+
+        _addr = hex_string_to_int(e.account)
+
+        if e.is_create:
+            _init_account(_addr)
+            accounts[_addr]['code'] = e.deployed_code
+
+        if e.updates_balance:
+            _init_account(_addr)
+            accounts[_addr]['balance'] = e.new_balance
+
+        for account_storage, slot, new_value in e.storage_updates:
+            _addr_account_storage = hex_string_to_int(account_storage)
+            _init_account(_addr_account_storage)
+            accounts[_addr_account_storage]['storage'][intToken(hex_string_to_int(slot))] = intToken(
+                hex_string_to_int(new_value)
+            )
+
+    return accounts
+
+
+def summary_to_account_cells(summary: dict) -> list[KApply]:
```

Comment:
This seems like the central feature in this PR, please add a (parametric) unit test that exercises it. This would catch changes to the `summary` schema.

Reply:
It would be good to convert `summary` into a statically typed data structure soon (or right) after reading the data from disk.

Reply:
Added a unit test in https://github.com/runtimeverification/kontrol/pull/271/commits/55155001459a65c4c4db8ef038cb7c8f76ddc8c6, and`summary` is now a `list[SummaryEntry]`.